Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs

```diff
@@ -142,20 +143,46 @@ public bool CanOpen(IEntity user)
             return accessReader.IsAllowed(user);
         }
 
-        public void TryOpen(IEntity user)
+        /// <summary>
+        /// Returns whether a door has a certain access type. For example, maintenance doors will have access type
+        /// "Maintenance" in their AccessReader.
+        /// </summary>
+        private bool HasAccessType(string accesType)
         {
-            if (!CanOpen(user))
+            if(Owner.TryGetComponent<AccessReader>(out var accessReader))
             {
-                Deny();
-                return;
+                return accessReader.AccessLists.Any(list => list.Contains(accesType));
             }
 
-            Open();
+            return false;
+        }
+
+        public void TryOpen(IEntity user)
+        {
+            var doorSystem = EntitySystem.Get<DoorSystem>();
+            var isAirlockExternal = HasAccessType("External");
 
-            if (user.TryGetComponent(out HandsComponent? hands) && hands.Count == 0)
+            var access = doorSystem.AccessType switch
             {
-                EntitySystem.Get<AudioSystem>().PlayFromEntity("/Audio/Effects/bang.ogg", Owner,
-                    AudioParams.Default.WithVolume(-2));
+                DoorSystem.AccessTypes.AllowAll => true,
+                DoorSystem.AccessTypes.AllowAllIdExternal => isAirlockExternal ? CanOpen(user) : true,
+                DoorSystem.AccessTypes.AllowAllNoExternal => !isAirlockExternal,
+                _ => CanOpen(user)
+            };
```

Comment:
The access type check should probably be moved into `CanOpen(IEntity)`, unless you specifically want that method to not affect overriding classes.

Reply:
Yes, that would make more sense. I'll move it.